src/qilib/configuration_helper/adapters/virtual_awg_instrument_adapter.py

```diff
@@ -0,0 +1,114 @@
+"""Quantum Inspire library
+
+Copyright 2019 QuTech Delft
+
+qilib is available under the [MIT open-source license](https://opensource.org/licenses/MIT):
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+try:
+    from qtt.instrument_drivers.virtualAwg.virtual_awg import VirtualAwg
```

Comment:
Here we have a circular reference: qtt depends on qilib, but in this PR you require qtt for qilib.

Since these are adapters for specific instruments, should not be part of qilib I think. Dynamically adding adapters (e.g. after qilib was imported) is possible, but perhaps a well-defined interface for that is needed.

@ammar92 @qSaevar

Reply:
Well, it only tries to import the virtual awg from qtt as we do not want/can't add qtt as a requirement due to circular reference as you mentioned.

To avoid this we created a special handeling for qtt instruments as can be seen with the time stamp instrument and the virtual dac instrument adapters.

Creating a nice way to dynamically adding adapters, as you suggest, might be a good idea. That, or adding all instruments to Qcodes, but that can be a bit of a hassle.